diag_manager/fms_diag_buffer.F90

```diff
@@ -42,25 +39,24 @@ module fms_diag_buffer_mod
 !> @brief Object that holds buffered data and other diagnostics
 !! Abstract to ensure use through its extensions(buffer0-5d types)
 type, abstract :: fmsDiagBuffer_class
-  class(*), dimension(:,:,:,:,:), allocatable :: remap_buffer !< remapped buffer data
+  class(*), dimension(:,:,:,:,:), pointer     :: remap_buffer !< remapped buffer data
```

Comment:
The buffer updater (the "averaging functions") in send_data_3d assumed the buffers are 4D (3D space + sample). Did we decide to move to 5D, or is this remapping for another purpose?